Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,322 @@
+            using VisitResponse = AZ::SettingsRegistryInterface::VisitResponse;
+
+            // optional default quality level
+            registry->GetObject(m_qualityLevel, FixedValueString::format("%s/Default",
+                m_path.c_str()));
+
+            // optional description
+            registry->Get(m_description, FixedValueString::format("%s/Description",
+                m_path.c_str()));
+
+            // count the number of defined quality levels for error checking
+            AZ::SettingsRegistryVisitorUtils::VisitArray(*registry,
+                [&]([[maybe_unused]] const auto& visitArrayArg)
+                {
+                    m_numQualityLevels++;
+                    return VisitResponse::Continue;
+                },
+                FixedValueString::format("%s/Levels", m_path.c_str()));
+        }
+
+        m_functor = AZStd::make_unique<QualityCVarGroupFunctor>(
+            m_name.c_str(),
+            m_description.c_str(),
+            AZ::ConsoleFunctorFlags::DontReplicate,
+            AZ::AzTypeInfo<QualityLevel>::Uuid(),
+            *this,
+            &QualityCVarGroup::CvarFunctor);
+    }
+
+    QualityCVarGroup::~QualityCVarGroup() = default;
+
+    inline void QualityCVarGroup::CvarFunctor(const AZ::ConsoleCommandContainer& arguments)
+    {
+        StringToValue(arguments);
+    }
+
+    QualityLevel QualityCVarGroup::GetQualityLevel(const AZ::CVarFixedString& value) const
+    {
+        // We don't use AZ::ConsoleTypeHelpers::ToValue because it will print
+        // a warning when it fails to convert a string to int32_t
+        // The QualityLevel strings are user-defined and not known at compile
+        // time so only the pre-defined values can be converted.
+
+        // handle integer values (fast path)
+        QualityLevel qualityLevel = FromNumber(value);
+
+        if (qualityLevel == QualityLevel::Invalid)
+        {
+            // handle enum pre-defined values 
+            qualityLevel = FromEnum(value);
+        }
+
+        if (qualityLevel == QualityLevel::Invalid)
+        {
+            // handle user-defined level name values
+            qualityLevel = FromName(value);
+        }
+
+        return qualityLevel;
+    }
+
+    QualityLevel QualityCVarGroup::FromNumber(const AZ::CVarFixedString& value) const
+    {
+        char* endPtr = nullptr;
+        int32_t intValue = aznumeric_cast<int32_t>(strtoll(value.c_str(), &endPtr, 0));
+
+        // don't allow values higher than the user-defined levels
+        if (endPtr == value.c_str() || intValue >= m_numQualityLevels)
+        {
+            return QualityLevel::Invalid;
+        }
+
+        return QualityLevel(intValue);
+    }
+
+    QualityLevel QualityCVarGroup::FromEnum(const AZ::CVarFixedString& value) const
+    {
+        using EnumMemberPair = typename AZ::AzEnumTraits<QualityLevel>::MembersArrayType::value_type;
+        auto findEnumOptionValue = [firstArg = value](EnumMemberPair enumPair)

```

Comment:
You can capture the `value` variable by reference and that would prevent an unnecessary string copy.
```suggestion
        auto findEnumOptionValue = [firstArg = &value](EnumMemberPair enumPair)

```